File: pearls/col1/prob3.go
package main

import (
    "os"
    "fmt"
    "./bitvec"
)

const MAX = 10000000

func main() {
    vec := bitvec.CreateBitVector(MAX)
    max := uint(0)
    for {
        var num uint
        n, err := fmt.Scan(&num)
        if err == os.EOF { break }
        if n > 0 {
            vec.SetBit(num)
            if num > max { max = num }
        }
    }
    for i := uint(0); i <= max; i++ {
        if val, _ := vec.GetBit(i); val > 0 {
            fmt.Println(i)
        }
    }
}

File: pearls/col1/java/prob3.java
import java.util.*;

public class prob3 {
    private static final int N = 10000000;
    public static void main(String[] args) {
	BitVector vec = new BitVector(N);
	Scanner scan = new Scanner(System.in);
	int max = 0;
	for (int n = 0; scan.hasNextInt(); n++) {
	    int num = scan.nextInt();
	    vec.setBit(num);
	    if (num > max) max = num;
	}
	for (int i = 0; i <= max; i++) {
	    if (vec.getBit(i) > 0) {
		System.out.println(i);
	    }
	}
    }
}

